fili-core/src/test/groovy/com/yahoo/bard/webservice/util/GroovyTestUtils.groovy

```diff
@@ -190,4 +192,46 @@ abstract class GroovyTestUtils extends Specification {
                 .collect {it.split("; rel=")} //Split into URL/rel pairs
                 .collectEntries { [(it[1].trim()): it[0].trim()] } //Trim the pairs and put them into a rel/url map
     }
+
+    /**
+     * A special-purpose operation that verifies that an error response is correct.
+     * <p>
+     * This function expects JSON Strings. It filters out both context from the Druid query contained in the error
+     * message (because it contains an untestable UUID for the query) and it only verifies that the queryId is present,
+     * since that is also an untestable UUID.
+     *
+     * @param errorBody  The error generated by the test
+     * @param expectedBody  The expected error body
+     */
+    static boolean compareErrorPayload(String errorBody, String expectedBody) {
+        Map errorBodyJson = JSON_PARSER.parseText(errorBody)
+        Map expectedBodyJson = JSON_PARSER.parseText(expectedBody)
+
+        // Using for loops for better error messages.
+        for (item in errorBodyJson) {
+            String key = item.key
+            def value = key == "druidQuery" ?
+                    //Empty out the query ID included in the context, since it is a randomly generated UUID
+                    item.value?.collectEntries {stripQueryId(it)} :
+                    item.value
+            assert value == expectedBodyJson[item.key] || (item.key == "requestId" && item.value)
+        }
+        for (item in expectedBodyJson) {
+            assert item.key in errorBodyJson.keySet()
+        }
+        return true
+    }
+
+    /**
+     * Given an entry in a map, returns either a copy of the entry, or a copy of the entry where its value does not
+     * contain the queryId key, if the entry's key is "context."
+     *
+     * @param entry  The entry to be transformed
+     *
+     * @return  Either a shallow copy of the entry, or a version of the entry where its value does not contain the
+     * "queryId" key.
+     */
+    private static Map stripQueryId(def entry) {
+        return [(entry.key): entry.key == "context" ? entry.value.findAll{ it.key != "queryId"} : entry.value]
+    }
```

Comment:
😲  I can't believe an argument that stemmed primarily from laziness actually convinced you. 

Be honest with me. It's because you like how groovy that code there is (well, except for the for-loop maybe), and you know that if I rewrite it to use an `ObjectMapper` it'll look all Javay and gross.